setup.py

```diff
@@ -20,6 +20,28 @@
 from setuptools import setup, Extension, find_packages
 from setuptools.command.build_ext import build_ext
 
+default_backend = "lightning_qubit"
+supported_backends = ("lightning_kokkos", "lightning_qubit")
+
+def get_backend():
+    backend = None
+    if "PL_BACKEND" in os.environ:
+        backend = os.environ.get("PL_BACKEND", default_backend)
+    if "CMAKE_ARGS" in os.environ:
+        cmake_args = os.environ["CMAKE_ARGS"].split(" ")
+        arg = [x for x in cmake_args if "PL_BACKEND" in x]
+        cmake_backend = arg[0].split("=")[1] if arg else default_backend
+        if backend is not None and backend != cmake_backend:
+            raise ValueError(f"Backends {backend} and {cmake_backend} specified by PL_BACKEND and CMAKE_ARGS respectively do not match.")
+        backend = cmake_backend
+    if backend is None:
+        backend = default_backend
+    if backend not in supported_backends:
+        raise ValueError(f"Invalid backend {backend}.")
+    return backend
```

Comment:
Hey @vincentmr. It seems like installation fails when you provide both PL_BACKEND and CMAKE_ARGS.
Examples: `PL_BACKEND="lightning_kokkos" CMAKE_ARGS="-DENABLE_PYTHON=ON" pip install -e .`

Reply:
Also, what do you think about making both `lightning_qubit` and `lightning.qubit` (same for kokkos) a valid option for PL_BACKEND?

Reply:
I'll fix it. 

> Also, what do you think about making both lightning_qubit and lightning.qubit (same for kokkos) a valid option for PL_BACKEND?

I'm good with that